keycloak/lib/all-stacks.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import { App, Stack, StackProps } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { VpcStack } from './stacks/vpc';
+import { KeycloakUtils } from './stacks/utils';
+import { RdsStack } from './stacks/rds';
+import { KeycloakStack } from './stacks/keycloak';
+import { KeycloakWAF } from './waf';
+
+export class AllStacks extends Stack {
+  static readonly HOSTED_ZONE = 'keycloak.opensearch.org'
+
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+    const app = new App();
```

Comment:
why is app getting initialized twice, once in bin and here?
This logic can be implemented in `bin/keycloak.ts` to avoid duplication.

Reply:
Good catch! Modified it.

Reply:
So we need to avoid nested stacks. One is the main entrypoint which calls other app. Else it created nested stacks.
@prudhvigodithi can add more.

Reply:
There should be a single root (app) that is a collection of all the stacks for single application. https://docs.aws.amazon.com/cdk/v2/guide/apps.html#apps-tree
A lot depends on what is the definition of application for you.